apollo-federation/src/operation/rebase.rs

```diff
@@ -288,26 +288,24 @@ impl FieldSelection {
         &self,
         parent_type: &CompositeTypeDefinitionPosition,
         schema: &ValidFederationSchema,
-    ) -> bool {
-        if &self.field.data().schema == schema
-            && parent_type == &self.field.data().field_position.parent()
+    ) -> Result<bool, FederationError> {
+        if self.field.data().schema == *schema
+            && self.field.data().field_position.parent() == *parent_type
         {
-            return true;
+            return Ok(true);
         }
 
-        let Some(ty) = self.field.type_if_added_to(parent_type, schema) else {
-            return false;
+        let Some(ty) = self.field.type_if_added_to(parent_type, schema)? else {
+            return Ok(false);
         };
 
         if let Some(set) = &self.selection_set {
-            if set.type_position != ty {
-                return set
-                    .selections
-                    .values()
-                    .all(|sel| sel.can_add_to(parent_type, schema));
+            let ty: CompositeTypeDefinitionPosition = ty.try_into()?;
```

Comment:
I think `type_if_added_to` already returns `CompositeTypeDefinitionPosition` so why do we need the extra `try_into`?

Reply:
nvm you changed the method return type 👍

Reply:
The return type of `Field.type_if_added_to()` was previously `CompositeTypeDefinitionPosition`, but this was causing it to return `None` for enums/scalars, which caused `FieldSelection.can_add_to()` to return `false`. I've updated it in this PR to return an `OutputTypeDefinitionPosition` instead, and only try converting to `CompositeTypeDefinitionPosition` when there's a selection set on the field.